File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/socialmedia/NotifyDeleteChirp.java
package com.github.dedis.popstellar.model.network.method.message.data.socialmedia;

import androidx.annotation.NonNull;

import com.github.dedis.popstellar.model.Immutable;
import com.github.dedis.popstellar.model.network.method.message.data.*;
import com.github.dedis.popstellar.model.objects.security.MessageID;
import com.google.gson.annotations.SerializedName;

@Immutable
public class NotifyDeleteChirp extends Data {

  @SerializedName("chirp_id")
  private final MessageID chirpId;

  private final String channel;
  private final long timestamp;

  /**
   * @param chirpId message ID of the chirp
   * @param channel channel where the post is located
   * @param timestamp UNIX timestamp in UTC
   */
  public NotifyDeleteChirp(MessageID chirpId, String channel, long timestamp) {
    this.chirpId = chirpId;
    this.channel = channel;
    this.timestamp = timestamp;
  }

  @Override
  public String getObject() {
    return Objects.CHIRP.getObject();
  }

  @Override
  public String getAction() {
    return Action.NOTIFY_DELETE.getAction();
  }

  public MessageID getChirpId() {
    return chirpId;
  }

  public String getChannel() {
    return channel;
  }

  public long getTimestamp() {
    return timestamp;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    NotifyDeleteChirp that = (NotifyDeleteChirp) o;
    return java.util.Objects.equals(getChirpId(), that.getChirpId())
        && java.util.Objects.equals(getChannel(), that.getChannel())
        && java.util.Objects.equals(getTimestamp(), that.getTimestamp());
  }

  @Override
  public int hashCode() {
    return java.util.Objects.hash(getChirpId(), getChannel(), getTimestamp());
  }

  @NonNull
  @Override
  public String toString() {
    return "NotifyDeleteChirp{"
        + "chirpId='"
        + chirpId.getEncoded()
        + '\''
        + ", channel='"
        + channel
        + '\''
        + ", timestamp='"
        + timestamp
        + '\''
        + '}';
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/socialmedia/NotifyDeleteChirp.kt
package com.github.dedis.popstellar.model.network.method.message.data.socialmedia

import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.network.method.message.data.Action
import com.github.dedis.popstellar.model.network.method.message.data.Data
import com.github.dedis.popstellar.model.network.method.message.data.Objects
import com.github.dedis.popstellar.model.objects.security.MessageID
import com.google.gson.annotations.SerializedName

@Immutable
class NotifyDeleteChirp
/**
 * @param chirpId message ID of the chirp
 * @param channel channel where the post is located
 * @param timestamp UNIX timestamp in UTC
 */
(
    @field:SerializedName("chirp_id") val chirpId: MessageID,
    val channel: String,
    val timestamp: Long
) : Data() {

  override val `object`: String
    get() = Objects.CHIRP.`object`

  override val action: String
    get() = Action.NOTIFY_DELETE.action

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val that = other as NotifyDeleteChirp
    return chirpId == that.chirpId && channel == that.channel && timestamp == that.timestamp
  }

  override fun hashCode(): Int {
    return java.util.Objects.hash(chirpId, channel, timestamp)
  }

  override fun toString(): String {
    return "NotifyDeleteChirp{chirpId='${chirpId.encoded}', channel='$channel', timestamp='$timestamp'}"
  }
}
